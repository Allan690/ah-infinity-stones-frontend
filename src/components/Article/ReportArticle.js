/* eslint-disable rule camelcase */
/* eslint-disable camelcase */
import React, { Component } from 'react';
import PropTypes from 'prop-types';

class ReportArticle extends Component {
  constructor(props) {
    super(props);
    this.handleReport = this.handleReport.bind(this);
    this.state = {
      reportMessage: '',
    };
    this.onChange = this.onChange.bind(this);
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  handleReport() {
    const { reportMessage } = this.state;
    const {
      props: {
        article: { art_slug },
        reportArticleAction,
      },
    } = this.props;
    const message = {
      artSlug: art_slug,
      report_msg: reportMessage,
    };
    reportArticleAction(message);
  }

  render() {
    const { reportMessage } = this.state;
    const modalContent = (
      <div className='modal-dialog modal-dialog-centered' role='document'>
        <div className='modal-content'>
          <div className='modal-header'>
            <h5 className='modal-title'>Report Article</h5>
            <button type='button' className='close' data-dismiss='modal' aria-label='Close'>
              <span aria-hidden='true'>&times;</span>
            </button>
          </div>
          <div className='modal-body'>
            <form id='reportMessageForm'>
              <input type='radio' name='reportMessage' value='Spam' onClick={this.onChange} />
              Spam
              <br />
              <input
                type='radio'
                name='reportMessage'
                value='Harrassment'
                onClick={this.onChange}
              />
              Harrassment
              <br />
              <input
                type='radio'
                name='reportMessage'
                value='Rules Violation'
                onClick={this.onChange}
              />
              Rules Violation
            </form>
          </div>
          <div className='modal-footer'>
            <button type='button' className='btn btn-secondary' data-dismiss='modal'>
              Cancel
            </button>
            <button
              type='button'
              className='btn btn-danger'
              id='modalSubmitReport'
              data-dismiss='modal'
              disabled={!reportMessage}
              onClick={this.handleReport}
            >
              Report
            </button>
          </div>
        </div>
      </div>
    );
    return (
      <div
        className='modal fade'
        id='reportArticleModal'
        tabIndex='-1'
        role='dialog'
        aria-labelledby='reportArticleModalTitle'
        aria-hidden='true'
      >
        {modalContent}
      </div>
    );
  }
}

ReportArticle.propTypes = {
  props: PropTypes.objectOf(PropTypes.objectOf(PropTypes.any)).isRequired,
};

export default ReportArticle;
