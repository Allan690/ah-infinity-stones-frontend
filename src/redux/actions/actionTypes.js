export const REQUEST_LOGIN = 'REQUEST_LOGIN';
export const RECEIVE_LOGIN = 'RECEIVE_LOGIN';
export const LOGIN_ERROR = 'LOGIN_ERROR';
export const SIGNUP_REQUEST = 'SIGNUP_REQUEST';
export const SIGNUP_SUCCESS = 'SIGNUP_SUCCESS';
export const SIGNUP_FAILURE = 'SIGNUP_FAILURE';

export const SOCIAL_LOGIN_REQUEST = 'SOCIAL_LOGIN_REQUEST';
export const SOCIAL_LOGIN_SUCCESS = 'SOCIAL_LOGIN_SUCCESS';
export const SOCIAL_LOGIN_ERROR = 'SOCIAL_LOGIN_ERROR';


export const REQUEST_INCREMENT = 'REQUEST_INCREMENT';
export const RECEIVE_INCREMENT = 'RECEIVE_INCREMENT';

export const PASSWORD_RESET_START = 'PASSWORD_RESET_START';
export const PASSWORD_RESET_COMPLETE = 'PASSWORD_RESET_COMPLETE';
export const PASSWORD_RESET_ERROR = 'PASSWORD_RESET_ERROR';


export const PASSWORD_RESET_UPDATE_START = 'PASSWORD_RESET_UPDATE_START';
export const PASSWORD_RESET_UPDATE_COMPLETE = 'PASSWORD_RESET_UPDATE_COMPLETE';
export const PASSWORD_RESET_UPDATE_ERROR = 'ASSWORD_RESET_UPDATE_ERROR';
