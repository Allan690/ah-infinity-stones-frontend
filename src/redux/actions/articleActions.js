import {
  CREATE_ARTICLE, CREATE_ARTICLE_SUCCESS, CREATE_ARTICLE_FAILURE,
  GET_ALL_ARTICLES_REQUEST, GET_ALL_ARTICLES_SUCCESS, GET_ALL_ARTICLES_FAILURE,
  GET_ONE_ARTICLE_REQUEST, GET_ONE_ARTICLE_SUCCESS, GET_ONE_ARTICLE_FAILURE,
  EDIT_ARTICLE, UPDATE_ARTICLE, DELETE_ARTICLE,
} from './actionTypes';


export const createArticleAction = article => ({
  type: CREATE_ARTICLE,
  payload: article,
});
export const createArticleSuccessAction = article => ({
  type: CREATE_ARTICLE_SUCCESS,
  payload: article,
});
export const createArticleFailureAction = error => ({
  type: CREATE_ARTICLE_FAILURE,
  payload: error,
});

export const getAllArticlesRequestAction = () => ({
  type: GET_ALL_ARTICLES_REQUEST,
});

export const getAllArticlesSuccessAction = articles => ({
  type: GET_ALL_ARTICLES_SUCCESS,
  payload: [articles],
});

export const getAllArticlesFailureAction = error => ({
  type: GET_ALL_ARTICLES_FAILURE,
  payload: error,
});

export const getOneArticleRequestAction = artSlug => ({
  type: GET_ONE_ARTICLE_REQUEST,
  payload: artSlug,
});

export const getOneArticleSuccessAction = article => ({
  type: GET_ONE_ARTICLE_SUCCESS,
  payload: article,
});

export const getOneArticleFailureAction = error => ({
  type: GET_ONE_ARTICLE_FAILURE,
  payload: error,
});

export const editArticleAction = article => ({
  type: EDIT_ARTICLE,
  payload: article,
});

export const updateArticleAction = article => ({
  type: UPDATE_ARTICLE,
  payload: article,
});

export const deleteArticleAction = art_slug => ({
  type: DELETE_ARTICLE,
  payload: art_slug,
});
