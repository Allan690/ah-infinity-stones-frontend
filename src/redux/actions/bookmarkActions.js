import {
  BOOKMARK_REQUEST,
  BOOKMARK_SUCCESS,
  BOOKMARK_ERROR,
} from './actionTypes';

export const bookmarkRequest = slug => ({
  type: BOOKMARK_REQUEST,
  slug,
});

export const bookmarkSuccess = data => ({
  type: BOOKMARK_SUCCESS,
  data,
});

export const bookmarkError = error => ({
  type: BOOKMARK_ERROR,
  error,
});
