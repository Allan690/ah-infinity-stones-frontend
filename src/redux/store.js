/* eslint no-underscore-dangle: 0 */
/* eslint-disable no-undef */
import { applyMiddleware, createStore, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import allReducers from './reducers';
import rootSaga from './sagas';

// only activate redux devtools on the browser otherwise just compose
const composer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const composeEnhancers = composer;

const middleware = createSagaMiddleware();
const store = createStore(
  allReducers,
  composeEnhancers(applyMiddleware(middleware)),
);

export default store;

middleware.run(rootSaga);
