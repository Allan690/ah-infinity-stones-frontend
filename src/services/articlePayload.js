import { baseURL } from '../components/common/constants';


export async function createArticle(article) {
  const response = await fetch(`${baseURL}articles`, {
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-type': 'application/json',
      Authorization: `${localStorage.getItem('Token')}`,
    },
    body: JSON.stringify(article),
  });
  if (response.status === 201) {
    const data = await response.json();
    return data;
  }
  throw new Error(response.status);
}

export async function getAllArticles() {
  const response = await fetch(`${baseURL}articles`, {
    method: 'GET',
    Authorization: `${localStorage.getItem('Token')}`,
  });
  if (response.status === 200) {
    const data = await response.json();
    return [response, data];
  }
  throw new Error(response.status);
}
export async function getOneArticle(artSlug) {
  const response = await fetch(`${baseURL}articles/${artSlug}`, {
    method: 'GET',
    Authorization: `${localStorage.getItem('Token')}`,
  });
  if (response.status === 200) {
    const data = await response.json();
    return [response, data];
  }
  throw new Error(response.status);
}
export async function updateArticle(article) {
  const response = await fetch(`${baseURL}articles/${article.artSlug}`, {
    method: 'PUT',
    headers: {
      Accept: 'application/json',
      'Content-type': 'application/json',
      Authorization: `${localStorage.getItem('Token')}`,
    },
    body: JSON.stringify(article),
  });
  if (response.status === 200) {
    const data = await response.json();
    return data;
  }
  throw new Error(response.status);
}
export async function deleteArticle(artSlug) {
  const response = await fetch(`${baseURL}articles/${artSlug}`, {
    method: 'DELETE',
    headers: {
      Authorization: `${localStorage.getItem('Token')}`,
    },
  });
  if (response.status === 204) {
    return response;
  }
  throw new Error(response);
}
