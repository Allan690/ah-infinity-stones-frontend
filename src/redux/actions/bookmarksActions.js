import { REQUEST_BOOKMARKS, BOOKMARKS_SUCCESS, BOOKMARKS_FAILURE } from './actionTypes';

export const requestBookmarks = () => ({
  type: REQUEST_BOOKMARKS,
});

export const bookmarksSuccess = data => ({
  type: BOOKMARKS_SUCCESS,
  data,
});

export const bookmarksFailure = error => ({
  type: BOOKMARKS_FAILURE,
  error,
});
