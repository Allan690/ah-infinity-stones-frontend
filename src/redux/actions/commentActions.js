import { GET_COMMENTS, POST_COMMENT, GET_TOPIC } from './actionTypes';

export const getComments = () => ({
  type: GET_COMMENTS,
});

export const postComment = postData => ({
  type: POST_COMMENT,
  payload: postData,
});

export const getTopic = () => ({
  type: GET_TOPIC,
});
