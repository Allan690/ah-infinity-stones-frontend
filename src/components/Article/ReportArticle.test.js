/* eslint-disable no-unused-vars */
/* eslint-disable import/default */
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { call } from 'redux-saga/effects';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import store from '../../redux/store';
import * as types from '../../redux/actions/actionTypes';
import * as actions from '../../redux/actions/articleActions';
import reducer from '../../redux/reducers/articleReducer';
import { reportArticle } from '../../services/articlePayload';
import ReportArticle from './ReportArticle';

configure({ adapter: new Adapter() });
describe('Action creators', () => {
  let payload = {};
  beforeEach(() => {
    payload = {
      artSlug: 'promaster-is-back-from-shanghai-00rf0e0r09r0',
      report_msg: 'Spam',
    };
  });
  it('should create an action to report an article', () => {
    const expectedAction = {
      type: types.REPORT_ARTICLE,
      payload,
    };
    expect(actions.reportArticleAction(payload)).toEqual(expectedAction);
  });
});

describe('Test article reporting Api call', () => {
  let payload = {};
  let response = {};
  beforeEach(() => {
    payload = {
      artSlug: 'learn-react-in-two-months-3eaAFEAGEedese#',
      report_msg: 'Spam',
    };
    response = reportArticle(payload);
  });

  it('A promise is returned from article rating api call', () => {
    expect(response).toBeTruthy();
  });

  it('Promise has data', () => {
    response.then((objects) => {
      expect(objects).toHaveLength(1);
    });
  });
});
describe('Report Article component', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const initialState = {};
  let store = '';
  let component;
  const props = {
    onChange: jest.fn(),
    props: {},
    state: {},
    handleReport: jest.fn(),
  };

  beforeEach(() => {
    store = mockStore(initialState);
    component = shallow(<ReportArticle {...props} />);
  });

  it('article report component should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
  it('shoud render subcomponents', () => {
    expect(component.find('#reportArticleModal'));
  });
  it('has div elements', () => {
    expect(component.find('div')).toHaveLength(6);
  });
});
