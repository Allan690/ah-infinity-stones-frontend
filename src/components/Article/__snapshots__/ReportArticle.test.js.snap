// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Report Article component article report component should match snapshot 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="reportArticleModalTitle"
  className="modal fade"
  id="reportArticleModal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog modal-dialog-centered"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
        >
          Report Article
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <form
          id="reportMessageForm"
        >
          <input
            name="reportMessage"
            onClick={[Function]}
            type="radio"
            value="Spam"
          />
          Spam
          <br />
          <input
            name="reportMessage"
            onClick={[Function]}
            type="radio"
            value="Harrassment"
          />
          Harrassment
          <br />
          <input
            name="reportMessage"
            onClick={[Function]}
            type="radio"
            value="Rules Violation"
          />
          Rules Violation
        </form>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          type="button"
        >
          Cancel
        </button>
        <button
          className="btn btn-danger"
          data-dismiss="modal"
          disabled={true}
          id="modalSubmitReport"
          onClick={[Function]}
          type="button"
        >
          Report
        </button>
      </div>
    </div>
  </div>
</div>
`;
